plugins/inputs/redis_sentinel/redis_sentinel.go

```diff
@@ -0,0 +1,503 @@
+package redis_sentinel
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"net/url"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/go-redis/redis"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/common/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type RedisSentinel struct {
+	Servers  []string `toml:"servers"`
+	Password string   `toml:"password"`
+	tls.ClientConfig
+
+	clients []*RedisSentinelClient
+}
+
+type RedisSentinelClient struct {
+	sentinel *redis.SentinelClient
+	tags     map[string]string
+}
+
+const measurementMasters = "redis_sentinel_masters"
+const measurementSentinel = "redis_sentinel"
+const measurementSentinels = "redis_sentinel_sentinels"
+const measurementReplicas = "redis_sentinel_replicas"
+
+func init() {
+	inputs.Add("redis_sentinel", func() telegraf.Input {
+		return &RedisSentinel{}
+	})
+}
+
+func (r *RedisSentinelClient) baseTags() map[string]string {
+	return r.tags
+}
```

Comment:
I think you should directly access the tags instead of defining this function.

Reply:
@srebhan Indded. Removed it in https://github.com/influxdata/telegraf/pull/10042/commits/c6b75d320996bdf1570a3ba8f623ed8b3e098514.